pkg/upgradecheck/upgrade_check.go

```diff
@@ -0,0 +1,143 @@
+package upgradecheck
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/instaclustr/operator/pkg/models"
+	"github.com/instaclustr/operator/pkg/scheduler"
+)
+
+func StartUpgradeCheckJob(ctx context.Context, client client.Client, s scheduler.Interface) error {
+	job := newUpgradeCheckJob(ctx, client)
+	return s.ScheduleJob(models.OperatorUpgradeChecker, models.UpgradeCheckInterval, job)
+}
+
+func newUpgradeCheckJob(ctx context.Context, client client.Client) scheduler.Job {
+	l := log.FromContext(ctx, "components", "WatchStatusJob")
+
+	return func() error {
+		// TODO: change from dockerhub to custom endpoint
+		latestTag, err := GetLatestDockerImageTag("icoperator/instaclustr-operator")
+		if err != nil {
+			l.Error(err, "unable to get latest docker image tag")
+		}
+
+		err = UpdateImageTagIfNeeded(ctx, l, client, latestTag)
+		if err != nil {
+			l.Error(err, "unable to update current image tag")
+		}
+
+		return nil
+	}
+}
+
+func GetLatestDockerImageTag(imageName string) (string, error) {
+	url := fmt.Sprintf("https://registry.hub.docker.com/v2/repositories/%s/tags?page_size=1", imageName)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
Let's use `io.ReadAll()` here and in other cases, cause the `ioutil` package is deprecated

Reply:
Fixed